### DISCUSSION TITLE
enviar email utilizando o XAMPP

### DISCUSSION THREAD

**Discussion by lnovaes** [RamonSilva20/mapos#1527] [General]
Prezados,
Vi que tem que configurar  cron jobs para enviar emails pendentes. Tem algum outra forma de eu enviar email utilizando o XAMPP?

**Comment by SuperPorte** [comment]
Saudações!!
Já eu nem sei onde fica esse cron jobs.
Onde encontro um manual para configurar isso?
Obrigado.

**Reply by RamonSilva20** [reply]
Olá @bernarcom, cron jobs seria no caso de sistemas operacionais Linux. Em distribuições Debian digitando "crontab -e" e adicionar os comandos de envio, apontando o caminho correto do sistema -> */2 * * * * php /var/www/index.php email/process
No caso de Windows provavelmente usando Tarefas Agendadas (Não tenho certeza).

**Comment by SuperPorte** [comment]
Mas estou no Linux mesmo.Windows pra mim é só vídeo game hehehe.Mas confesso ser um zero a esquerda em códigos.Estou a meses tentando usar o mapos lá na minha oficina de computadores mas estou apanhando muito.Nunca consegui por em rede e nem mandar e-mail.Esse crontab é só digitar no terminal e pronto?Muitíssimo obrigado e muito sucesso!!Em 10 de nov de 2021 16:49, Ramon Silva ***@***.***> escreveu:

Olá @bernarcom, cron jobs seria no caso de sistemas operacionais Linux. Em distribuições Debian digitando "crontab -e" e adicionar os comandos de envio, apontando o caminho correto do sistema -> */2 * * * * php /var/www/index.php email/process

No caso de Windows provavelmente usando Tarefas Agendadas (Não tenho certeza).

**Comment by SuperPorte** [comment]
![image](https://user-images.githubusercontent.com/68247011/141516508-f5a39223-308e-4dcb-8971-6af2a0a7be91.png)
Era isso mesmo pra fazer?

**Reply by RamonSilva20** [reply]
Isso mesmo, se o caminho do php e do sistema estiverem corretos já deve funcionar. Lembrando que também é preciso configurar as credenciais do email que usará para fazer os envios.

**Reply by SuperPorte** [reply]
Caminho PHP que você diz é aonde está o htdocs?

**Reply by SuperPorte** [reply]
![image](https://user-images.githubusercontent.com/68247011/141517816-8196b95b-da9c-4b67-8bd9-296c2f2484c6.png)
eu teria que mudar qual caminho?

**Reply by SuperPorte** [reply]
olhando na árvore de pastas, no meu linux mint, não tem nenhuma das duas estruturas de pastas.
wi@lab:/usr/local/bin$ dir
apt	    highlight-mint	 mint-sha256sum  woeusb     yelp
gnome-help  mate-system-monitor  search		 woeusbgui

wi@lab:/var$ dir
backups  cache	lib  local  lock  log  mail  opt  run  spool  tmp
wi@lab:/var$ 

E agora?

**Reply by RamonSilva20** [reply]
Use o comando "whereis php" para descobrir o caminho do PHP.

**Reply by SuperPorte** [reply]
wi@lab:~$ whereis php
php: /opt/lampp/bin/php

Troquei o caminhho no cron
Mas o segundo caminho "/var/www/html/index.php" ,ão tem no meu sistema.
Preciso fazer mais alguma coisa? Valeu!!!

**Reply by RamonSilva20** [reply]
O segundo caminho é onde o mapos está. 

Exemplo: /opt/lampp/htdocs/mapos/index.php

**Reply by SuperPorte** [reply]
Opa. Desculpa o sumiço. Meu filho veio pra minha casa e aí já era trabalho...
Vamos lá:
`wi@lab:/opt/lampp$ whereis php
php: /opt/lampp/bin/php
wi@lab:/opt/lampp$ 
`
crontab -e
`# m h  dom mon dow   command
*/2 * * * * /opt/lampp/bin/php /var/www/html/index.php email/process >> /var/log/cron.log
*/5 * * * * /opt/lampp/bin/php /var/www/html/index.php email/retry >> /var/log/cron.log
`
htdocs/mapos/application/config/mail.php
`<?php

$config['protocol']         = 'smtp';
$config['smtp_host']        = 'smtp.mail.com';
$config['smtp_crypto']      = 'ssl'; // tls or ssl
$config['smtp_port']        = 587;
$config['smtp_user']        = 'bernarcom@mail.com';
$config['smtp_pass']        = 'minhasenha';
$config['validate']         = true; // validar email
$config['mailtype']         = 'html'; // text ou html
$config['charset']          = 'utf-8';
$config['newline']          = "\r\n";
$config['bcc_batch_mode']   = false;
$config['wordwrap']         = false;
$config['priority']         = 3; // 1, 2, 3, 4, 5 | Email Priority. 1 = highest. 5 = lowest. 3 = normal.
`

Ainda não vai.
O que mais eu devo olhar?

**Comment by Pr3d4dor** [comment]
@bernarcom Salvou alguma coisa nos logs?

**Comment by SuperPorte** [comment]
Não sei dizer 😔Tem de fazer algo para salvar ou são criados automático?Se tiver de fazer algo, o que devo fazer?Em 17 de nov de 2021 20:07, Gianluca Bine ***@***.***> escreveu:
@bernarcom Salvou alguma coisa nos logs?

**Comment by SuperPorte** [comment]
meu cron está assim:
*/2 * * * * /opt/lampp/bin/php /opt/lampp/htdocs/mapos/index.php email/process >> /var/log/cron.log
*/5 * * * * /opt/lampp/bin/php /opt/lampp/htdocs/mapos/index.php email/retry >> /var/log/cron.log
Como root, fui na pasta /var/log/ e ví que não havia um arquivo cron.log e criei ele na pasta
Ainda assim, não escreve nada no arquivo log criado e não manda email.
Onde mais devo verificar?

**Comment by Pr3d4dor** [comment]
@bernarcom Os logs estão sendo salvos no arquivo /var/log/cron.log, se puder olhar ele.

**Reply by SuperPorte** [reply]
Vish, então não estão gerando.
ontem eu olhei nessa pasta e o arquivo não existia. Então eu mesmo criei um lá. Mas contunia em branco :(

**Reply by SuperPorte** [reply]
Continua em branco. Deletei o arquivo que criei. Vai que esteja atrapalhando né...

**Reply by Pr3d4dor** [reply]
@bernarcom Então o seu cron na verdade não está funcionando, tente entrar em contato com a hospedagem.

**Reply by SuperPorte** [reply]
Então.... hehehe.... Minha hospedagem sou eu mesmo.
Estou rodando o mapos no meu PC com linux lmde4 (linux mint debian edition)
Como sou uma oficina de porta pra rua, não posso ter o luxo de ter windows rodando no servidor. Aparece cada virus aqui que se eu usasse windows teria de formatar meus pcs todo dia. hehehe

**Comment by SuperPorte** [comment]
Acabei de consegui enviar email!!!!
Não sei o por que mas o TERMINAL MATE não segurou as atualizações feitas no crontab -e
quando abrir ocm o TERMINAL GNOME, deu certo de primeira!!!!

**Reply by SuperPorte** [reply]
Alguém conseguiu fazer no windows?
